src/library-aux/scala/Any.scala

```diff
@@ -138,13 +138,87 @@ abstract class Any {
    */
   final def isInstanceOf[T0]: Boolean = sys.error("isInstanceOf")
 
-  /** Cast the receiver object to be of type `T0`.
-   *
-   *  Note that the success of a cast at runtime is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.asInstanceOf[String]` will throw a `ClassCastException` at
-   *  runtime, while the expression `List(1).asInstanceOf[List[String]]` will not.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the requested type.
+  /** Tell the compiler to skip type safety and treat the receiver object to be of type `T0`.
+   *
+   *  This method is intended for cases where the compiler isn't able to know or capable to infer `T0` and neither
+   *  is it possible to check the subtype relation at runtime and hence skipping type safety is the only option.
+   *
+   *  DON'T USE `asInstanceOf` for type checking, it is **not** an alias for something like
```

Comment:
```suggestion
   *  DON'T USE `asInstanceOf` for type testing, it is **not** an alias for something like
```
In general, we use *checking* when we always expect that the check succeeds (and therefore failing the *check* is an *error*). We use *testing* when both "yes" and "no" are expected, correct answers.

Reply:
I disagree in this case, the user may assume that `foo.asInstanceOf[List[String]]` will fail if the underlying type of `foo` is `List[Int]`, and is a safe check. but this isn't a valid way to check, as it will succeed at runtime

Reply:
I think we see it the same way. type testing returns a boolean and it is up to the user what he does with this result whereas type checking errors at runtime if the type test returns false. Now, given that asInstanceOf does not throw in call cases where type testing returns false, we should emphasise that asInstanceOf is not intended for type checking.

Reply:
No it's fine. Leave it as is.